airflow/www/views.py

```diff
@@ -3194,16 +3194,22 @@ def action_has_dag_edit_access(action_func: Callable) -> Callable:
     @wraps(action_func)
     def check_dag_edit_acl_for_actions(
         self,
-        items: Optional[Union[List[TaskInstance], List[DagRun], TaskInstance, DagRun, None]],
+        items: Optional[Union[Iterable[TaskInstance], Iterable[DagRun], TaskInstance, DagRun]],
```

Comment:
The implementation currently can only take `list` (aside from one single TaskInstance and DagRun), if you change this, you need to rewrite the implementation as well.